tools/check_volume_levels.m

```diff
@@ -251,3 +279,30 @@ function plot_levels(meas, tc, lm)
 		y(:,j) = filter(b, a, x(:,j));
 	end
 end
+
+% This function is copy of
+% sof/tools/test/audio/test_utils/multitone.m
+function x = multitone( fs, f, amp, tlength )
+	n = round(fs*tlength);
+	t = (0:n-1)/fs;
+	nf = length(f);
+	if nf > 1
+		ph = rand(nf, 1)*2*pi;
+	else
+		ph = 0;
+	end
+
+	x = zeros(n, 1);
+	x(:,1) = amp(1)*sin(2*pi*f(1)*t);
+	for i=2:length(f)
```

Comment:
Why is the loop initialized from 2 rather than 1?

Reply:
It's because the first frequency f(1) sine wave is mixed before the loop. Any additional frequencies f(2) to f(end) are mixed in the loop. But yes, it can be simplified to e.g. 

```
	x = zeros(n, 1);
	for i=1:length(f)
		x(:,1) = x(:,1) + amp(i)*sin(2*pi*f(i)*t+ph(i))';
	end

```